bindings/cpp/src/Session.hpp

```diff
@@ -191,47 +168,45 @@ class Subscribe
 public:
     /** Wrapper for [sr_subscription_ctx_t](@ref sr_subscription_ctx_t) */
     Subscribe(S_Session sess);
-
     /** Wrapper for [sr_module_change_subscribe](@ref sr_module_change_subscribe) */
-    void module_change_subscribe(const char *module_name, S_Callback callback, const char *xpath = nullptr, \
-            void *private_data = nullptr, uint32_t priority = 0, sr_subscr_options_t opts = SUBSCR_DEFAULT);
+    void module_change_subscribe(const char *module_name, ModuleChangeCb cb, const char *xpath = nullptr, uint32_t priority = 0, sr_subscr_options_t opts = SUBSCR_DEFAULT);
     /** Wrapper for [sr_rpc_subscribe](@ref sr_rpc_subscribe) */
-    void rpc_subscribe(const char *xpath, S_Callback callback, void *private_data = nullptr, uint32_t priority = 0, \
-            sr_subscr_options_t opts = SUBSCR_DEFAULT);
+    void rpc_subscribe(const char *xpath, RpcCb cb, uint32_t priority = 0, sr_subscr_options_t opts = SUBSCR_DEFAULT);
     /** Wrapper for [sr_rpc_subscribe_tree](@ref sr_rpc_subscribe_tree) */
-    void rpc_subscribe_tree(const char *xpath, S_Callback callback, void *private_data = nullptr, uint32_t priority = 0, \
-            sr_subscr_options_t opts = SUBSCR_DEFAULT);
+    void rpc_subscribe_tree(const char *xpath, RpcTreeCb cb, uint32_t priority = 0, sr_subscr_options_t opts = SUBSCR_DEFAULT);
     /** Wrapper for [sr_event_notif_subscribe](@ref sr_event_notif_subscribe) */
-    void event_notif_subscribe(const char *module_name, S_Callback callback, const char *xpath = nullptr, \
-            time_t start_time = 0, time_t stop_time = 0, void *private_data = nullptr, sr_subscr_options_t opts = SUBSCR_DEFAULT);
+    void event_notif_subscribe(const char *module_name, EventNotifCb cb, const char *xpath = nullptr, time_t start_time = 0, time_t stop_time = 0, sr_subscr_options_t opts = SUBSCR_DEFAULT);
     /** Wrapper for [sr_event_notif_subscribe_tree](@ref sr_event_notif_subscribe_tree) */
-    void event_notif_subscribe_tree(const char *module_name, S_Callback callback, const char *xpath = nullptr, \
-            time_t start_time = 0, time_t stop_time = 0, void *private_data = nullptr, sr_subscr_options_t opts = SUBSCR_DEFAULT);
+    void event_notif_subscribe_tree(const char *module_name, EventNotifTreeCb cb, const char *xpath = nullptr, time_t start_time = 0, time_t stop_time = 0, sr_subscr_options_t opts = SUBSCR_DEFAULT);
     /** Wrapper for [sr_oper_get_items_subscribe](@ref sr_oper_get_items_subscribe) */
-    void oper_get_items_subscribe(const char *module_name, const char *path, S_Callback callback, \
-            void *private_data = nullptr, sr_subscr_options_t opts = SUBSCR_DEFAULT);
-    std::vector<S_Callback > cb_list;
+    void oper_get_items_subscribe(const char *module_name, OperGetItemsCb cb, const char *path, sr_subscr_options_t opts = SUBSCR_DEFAULT);
 
     /** Wrapper for [sr_get_event_pipe](@ref sr_get_event_pipe) */
```

Comment:
this method can be removed now; in case an evil user really needs this, they can use that private API (`swig_sub()`)

Reply:
done